terragrunt/modules/dev-desktops-azure/firewall.tf

```diff
@@ -0,0 +1,74 @@
+resource "azurerm_network_security_group" "dev_desktops" {
```

Comment:
Nit: I think some of these modules could use some comments with a short explainer about what they're doing.